server/src/main/java/org/opensearch/script/ScriptCache.java

```diff
@@ -130,6 +131,9 @@
             Throwable cause = executionException.getCause();
             if (cause instanceof ScriptException) {
                 throw (ScriptException) cause;
+            } else if (cause instanceof CircuitBreakingException) {
+                throw new OpenSearchRejectedExecutionException(
+                    "Failed to compile " + type + " script [" + id + "] using lang [" + lang + "]", cause);
```

Comment:
`CircuitBreakingException` already returns 429?

Reply:
`CircuitBreakingException` was wrapped in `GeneralScriptException` which return 500 error. You can have a look at reproduced [error](https://gist.github.com/shiv0408/02a56c4b8eb306f2781551521cdd4d12).

Reply:
Ideally the status is determined by the underlying cause https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/OpenSearchException.java#L253-L260. Can you check if there is something amiss here. In this case since

Also please update the steps to reproduce